### PR TITLE
copr: Vectorize time_to_sec

### PR DESCRIPTION
**Pull request by trabbart**
Signed-off-by: Bartosz Podkanowicz <podkanowicz.bartosz@gmail.com>

### What problem does this PR solve?

vectorize function time_to_sec

### Check List #5751 

Tests 

- Unit test

### CODE REVIEW

File: components/tidb_query_vec_expr/src/impl_time.rs [hunk 1]
```diff
@@ -172,6 +172,12 @@ pub fn second(t: Option<&Duration>) -> Result<Option<Int>> {
     Ok(t.as_ref().map(|t| i64::from(t.secs())))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn time_to_sec(t: Option<&Duration>) -> Result<Option<Int>> {
+    Ok(t.as_ref().map(|t| i64::from(t.to_secs())))
```
Comment (sticnarf) [inline components/tidb_query_vec_expr/src/impl_time.rs:178]: `i64::from` seems redundant.

File: components/tidb_query_vec_expr/src/impl_time.rs [hunk 2]
```diff
@@ -719,6 +725,40 @@ mod tests {
...
+        // test NULL case
+        let test_null_case = |sig| {
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(None::<Duration>)
+                .evaluate::<Int>(sig)
+                .unwrap();
+            assert_eq!(output, None);
+        };
+
+        test_null_case(ScalarFuncSig::Hour);
```
Comment (sticnarf) [inline components/tidb_query_vec_expr/src/impl_time.rs:759]: Why are we testing with `ScalarFuncSig::Hour`? And also `time_to_sec` needs a parameter.

Comment (trabbart) [reply components/tidb_query_vec_expr/src/impl_time.rs]: This line is redundant. I deleted it.

File: components/tidb_query_vec_expr/src/impl_time.rs (hunk 1 above)
Comment (trabbart) [reply components/tidb_query_vec_expr/src/impl_time.rs]: I deleted `i64::from`

File: components/tidb_query_vec_expr/src/impl_time.rs (hunk 2 above)
Comment (trabbart) [reply components/tidb_query_vec_expr/src/impl_time.rs]: It was mistake. I rewrited test.

**Pull request merged**